### ISSUE TITLE
Control Characters are not skipped with google_checks config

### ISSUE THREAD

**Issue by dimo414** [checkstyle/checkstyle#3700]
```
/tmp $ javac Demo.java

/tmp $ cat Demo.java
public class Demo {
  public static void main(String[] args) {
    System.out.println("\u0007");
  }
}

/tmp $ java -cp checkstyle-7.4-all.jar com.puppycrawl.tools.checkstyle.Main -c /google_checks.xml Demo.java 
Starting audit...
[WARN] /tmp/Demo.java:3: Unicode escape(s) usage should be avoided. [AvoidEscapedUnicodeCharacters]
Audit done.
```
---------------

Describe what you expect in detail.
I expected this file to pass with no warnings, as `\u0007` [`BEL` is a control character](http://www.fileformat.info/info/unicode/char/0007/index.htm), and google_checks.xml permits unicode escapes for such characters.

--------------

If I switch from  `\u0007` to `\u0008` [BACKSPACE](http://www.fileformat.info/info/unicode/char/0008/index.htm) I see a second, even more confusing error:

```
$ java -cp checkstyle-7.4-all.jar com.puppycrawl.tools.checkstyle.Main -c /google_checks.xml Demo.java 
Starting audit...
[WARN] /tmp/Demo.java:3: Unicode escape(s) usage should be avoided. [AvoidEscapedUnicodeCharacters]
[WARN] /tmp/Demo.java:3:24: Avoid using corresponding octal or Unicode escape. [IllegalTokenText]
Audit done.
```

It's not clear what I'm supposed to do here.

I'm happy to submit patches to address this, but I'm not certain what the expected behavior is.

**Comment by romani** [comment]
issue is confirmed:

Google style config has option "allowEscapesForControlCharacters=true" , https://github.com/search?q=path%3Asrc%2Fmain%2Fresources+filename%3Agoogle_checks.xml+repo%3Acheckstyle%2Fcheckstyle+AvoidEscapedUnicodeCharacters

smth wrong in regexp - https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java#L122

**Comment by rnveach** [comment]
@romani Should only 0007 be allowed as control character?
Is there someplace that lists all the control characters?

**Comment by romani** [comment]
@rnveach , 0007 should be enforced by UTs to show user that it will not happen again.
It will be good to recheck all other characters and enforce them by UTs too.

**Comment by Luolc** [comment]
I am on it :)

**Comment by romani** [comment]
fix is merged
